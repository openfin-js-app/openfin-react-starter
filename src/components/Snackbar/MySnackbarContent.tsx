import * as React from 'react';
import { MouseEventHandler } from "react";
import cx from 'classnames';

import SnackbarContent from '@material-ui/core/SnackbarContent';
import {SnackbarContentProps} from "@material-ui/core/SnackbarContent/SnackbarContent";
import IconButton from '@material-ui/core/IconButton';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import DesktopMacIcon from '@material-ui/icons/DesktopMac';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import CloseIcon from '@material-ui/icons/Close';
import WarningIcon from '@material-ui/icons/Warning';
import FaceIcon from '@material-ui/icons/Face'

import { WithStyles, withStyles} from '@material-ui/core/styles';
import { snackbarContentCompStyle as style } from '../../assets/jss/openfin-starter';

const variantIcon ={
    primary: DesktopMacIcon,
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
    rose: FaceIcon,
};

interface IMySnackbarContentProps{
    classes?:any,
    className?:string,
    message:string,
    onClose:MouseEventHandler<any>,
    variant:string,
}

type Props = IMySnackbarContentProps & WithStyles<typeof style> & SnackbarContentProps

class MySnackbarContentComp extends React.Component<Props,{}>{
    render(){

        const {
            classes, className, message, onClose, variant, ...other
        } = this.props;

        const Icon = variantIcon[variant];

        return(
            <SnackbarContent
                className={cx( classes.outmostContent, classes[variant],className)}
                aria-describedby={"client-snackbar"}
                message={
                    <span id={"client-snackbar"} className={classes.message}>
                        <Icon className={cx( classes.icon, classes.iconVariant )} />
                        {message}
                    </span>
                }
                action={
                    <IconButton
                        key={"close"}
                        aria-label={"Close"}
                        color={"inherit"}
                        className={classes.close}
                        onClick={onClose}
                    >
                        <CloseIcon className={classes.icon}/>
                    </IconButton>
                }
                {...other}
            />
        );
    }
}

export default withStyles(style)(MySnackbarContentComp);