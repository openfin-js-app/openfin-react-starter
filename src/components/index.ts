// BEGIN_OF_AL_CLI_COMPONENTS
export {default as ClientCounter} from './openfin-starter/client/ClientCounter';
export {default as Header} from './openfin-starter/Header/Header';
export {default as HeaderLinks} from './openfin-starter/Header/HeaderLinks';
export {default as HeaderThemeSwitcher } from './openfin-starter/Header/HeaderThemeSwitcher'
export {default as Sidebar} from './openfin-starter/Sidebar/Sidebar';
export {default as SnackbarContent} from './openfin-starter/Snackbar/MySnackbarContent';
export {default as OfflineOverlay} from './openfin-starter/Overlays/OfflineOverlay';
// END_OF_AL_CLI_COMPONENTS
