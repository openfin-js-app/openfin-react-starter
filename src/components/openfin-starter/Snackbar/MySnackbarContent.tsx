import * as React from 'react';
import { MouseEventHandler } from "react";
import cx from 'classnames';

import SnackbarContent from '@material-ui/core/SnackbarContent';
import {SnackbarContentProps} from "@material-ui/core/SnackbarContent/SnackbarContent";
import IconButton from '@material-ui/core/IconButton';

import CheckCircleIcon from '@material-ui/icons/CheckCircle';
import DesktopMacIcon from '@material-ui/icons/DesktopMac';
import ErrorIcon from '@material-ui/icons/Error';
import InfoIcon from '@material-ui/icons/Info';
import CloseIcon from '@material-ui/icons/Close';
import WarningIcon from '@material-ui/icons/Warning';
import FaceIcon from '@material-ui/icons/Face'

import { makeStyles } from '@material-ui/styles';
import { snackbarContentCompStyle as style } from '../../../assets/jss/openfin-starter';

const variantIcon ={
    primary: DesktopMacIcon,
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
    rose: FaceIcon,
};

interface IMySnackbarContentProps{
    classes?:any,
    className?:string,
    message:string,
    onClose:MouseEventHandler<any>,
    variant:string,
}

type Props = IMySnackbarContentProps & SnackbarContentProps

const useStyles = makeStyles(style);

const MySnackbarContentComp:React.FunctionComponent<Props> = (
    {
        className, message, onClose, variant, ...other
    }
)=>{

    const classes = useStyles();

    const Icon = variantIcon[variant];

    return(
        <SnackbarContent
            className={cx( classes.outmostContent, classes[variant],className)}
            aria-describedby={"client-snackbar"}
            message={
                <span id={"client-snackbar"} className={classes.message}>
                        <Icon className={cx( classes.icon, classes.iconVariant )} />
                    {message}
                    </span>
            }
            action={
                <IconButton
                    key={"close"}
                    aria-label={"Close"}
                    color={"inherit"}
                    onClick={onClose}
                >
                    <CloseIcon className={classes.icon}/>
                </IconButton>
            }
            {...other}
        />
    );

}

export default MySnackbarContentComp;