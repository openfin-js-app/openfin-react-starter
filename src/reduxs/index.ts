import {combineReducers} from 'redux';
import {IApplicationState} from './application/types';
import {IClientState} from './client/types';
import {IConfigState} from './config/types';

import application from './application/reducer';
import { defaultState as applicationDefaultState } from './application/reducer';
import client from './client/reducer';
import { defaultState as clientDefaultState } from './client/reducer';
import config from './config/reducer';
import { defaultState as configDefaultState } from './config/reducer';

export * from './application/actions';
export * from './application/types';
export * from './client/actions';
export * from './client/types';
export * from './config/actions';
export * from './config/context';
export * from './config/types';

export interface IRootState {
    application:IApplicationState,
    client:IClientState,
    config:IConfigState,
}

export const rootDefaultState = {
    application:applicationDefaultState,
    client:clientDefaultState,
    config:configDefaultState,
}

export default (parentWindowState?:IRootState)=>{
    if (parentWindowState){
        return combineReducers({
            application:application(parentWindowState.application),
            client:client(parentWindowState.client),
            config:config(parentWindowState.config),
        });

    }else{
        return combineReducers({
            application:application(),
            client:client(),
            config:config(),
        });
    }
}
