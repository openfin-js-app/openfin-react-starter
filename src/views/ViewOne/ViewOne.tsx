import * as React from 'react';

export default class ViewOne extends React.Component<any,any>{
    render(){
        return(
            <React.Fragment>
                <span>Viewone works</span>
            </React.Fragment>
        );
    }
}