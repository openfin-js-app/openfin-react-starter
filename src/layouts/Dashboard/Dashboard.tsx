import * as React from 'react';
import { connect } from 'react-redux';
import { Switch, Route, Redirect } from 'react-router-dom';
import cx from 'classnames';

import Snackbar from '@material-ui/core/Snackbar';

import { WithStyles, withStyles } from '@material-ui/core/styles';

import { Window } from '@albertli90/redux-openfin';

import {
    // acitons
    applicationDrawerToggle, applicationToogleWindowState,
    applicationSetSnackbarStatus, applicationProcessSnackbarQueue,
    applicationCloseSnackbar, applicationLaunchBarToggle,
    // types
    IRootState,ISnackBarMsg,
} from '../../redux';

import dashboardRoutes from '../../routes/Dashboard';
import { Sidebar, Header, SnackbarContent } from '../../components';

import { dashboardLayoutStyle as style } from '../../assets/jss/openfin-starter';

const switchRoutes = (
    <Switch>
        {dashboardRoutes.map((prop:any,key)=>{
            if (prop.redirect)
                return <Redirect from={prop.path} to={prop.to} key={key}/>;
            return <Route path={prop.path} component={prop.component} key={key}/>

        })}
    </Switch>
);

interface IProps extends WithStyles<typeof style>{
    drawerOpen:boolean,
    snackBarOpen:boolean,
    snackBarMsgInfo:Partial<ISnackBarMsg>,
    windowsState:string,
    actions:{
        handleDrawerToggle: ()=> void,
        handleSnackbarClose: (event:any,reason:string)=> void,
        handleSnackbarCloseBtnClick: ()=> void,
        handleSnackbarExited: ()=> void,
        handleSwitchToLaunchBar: ()=> void,
        handleMinimize: ()=> void,
        handleMaximize: ()=> void,
        handleClose: ()=> void,
    }
}

class DashbardLayout extends React.Component<IProps,{}>{
    render(){

        const {
            classes,
            drawerOpen,
            snackBarOpen, snackBarMsgInfo, windowsState,
            actions:{
                handleDrawerToggle,
                handleSnackbarClose, handleSnackbarCloseBtnClick, handleSnackbarExited,
                handleSwitchToLaunchBar, handleMinimize, handleMaximize, handleClose,
            },
            ...rest
        } = this.props;

        return(<React.Fragment>
            <Header
                routes={dashboardRoutes}
                windowsState={windowsState}
                handleDrawerToggle={handleDrawerToggle}
                open={drawerOpen}
                color={'primary'}
                onSwitchToLaunchBar={handleSwitchToLaunchBar}
                onMinimize={handleMinimize}
                onMaximize={handleMaximize}
                onClose = {handleClose}
                {...rest}
            />
            <div className={cx(classes.wrapper, 'primary-top-to-bottom')}>
                <Sidebar
                    routes={dashboardRoutes}
                    open={drawerOpen}
                    color={"primary"}
                    image={'/img/sidebar-1.jpg'}
                    {...rest}
                />
                <div className={cx(
                    classes.mainPanel,
                    {
                        [classes.mainPanelShift]:!drawerOpen
                    }
                )}
                >
                    <div className={classes.container}>
                        <div className={classes.content}>
                            {switchRoutes}
                        </div>
                    </div>
                </div>
            </div>
            {/*snackbar*/}
            <Snackbar
                key={snackBarMsgInfo.key}
                anchorOrigin={{
                    vertical:'bottom',
                    horizontal:'center'
                }}
                open={snackBarOpen}
                autoHideDuration={6000}
                onClose={handleSnackbarClose}
                onExited={handleSnackbarExited}
            >
                <SnackbarContent
                    onClose={handleSnackbarCloseBtnClick}
                    variant={snackBarMsgInfo.variant}
                    message={snackBarMsgInfo.message}
                />
            </Snackbar>
        </React.Fragment>);
    }
}

export default connect(
    (state:IRootState)=>({
        drawerOpen:state.application.drawerOpen,
        snackBarOpen:state.application.snackBarOpen,
        snackBarMsgInfo:state.application.snackBarMsgInfo,
        windowsState:state.application.windowsState,
    }),
    dispatch => ({
        actions:{
            handleDrawerToggle: ()=> {dispatch(applicationDrawerToggle());},
            handleSnackbarClose: (event,reason) => {dispatch(applicationCloseSnackbar({event,reason}))},
            handleSnackbarCloseBtnClick: () => {dispatch(applicationSetSnackbarStatus({open:false}))},
            handleSnackbarExited: () => {dispatch(applicationProcessSnackbarQueue())},
            // openfin
            handleSwitchToLaunchBar:()=>{dispatch(applicationLaunchBarToggle())},
            handleMinimize: ()=>{dispatch(Window.actions.minimize({}))},
            handleMaximize: ()=>{dispatch(applicationToogleWindowState())},
            handleClose:()=>{dispatch(Window.actions.close({force:false}))},
        }
    })
)(withStyles(style)(DashbardLayout));