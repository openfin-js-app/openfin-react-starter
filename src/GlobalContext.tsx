import * as React from 'react';
import { connect } from 'react-redux'

import { I18Language, IConfigRuntimeState } from './reduxs'

import { ConfigContextProvider } from './reduxs/config/context';

interface IProps {
    config:IConfigRuntimeState,
    onToggleThemeField?:()=>void,
    onUpdateLangField?:(lang:I18Language)=>void,
}

class GlobalContext extends React.Component<IProps,{}>{

    render(){

        const {
            config,
            onToggleThemeField,
            onUpdateLangField,
        } = this.props;

        return (
            <ConfigContextProvider value={
                {
                    config,
                    actions:{
                        onToggleThemeField,
                        onUpdateLangField,
                    }
                }
            }>
                {this.props.children}
            </ConfigContextProvider>

        )
    }
}

export default GlobalContext;