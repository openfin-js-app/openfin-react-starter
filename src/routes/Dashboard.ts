import AccessibilityIcon from '@material-ui/icons/Accessibility';
import BusinessIcon from '@material-ui/icons/Business';
import ChatIcon from '@material-ui/icons/Chat';
import ReportIcon from '@material-ui/icons/Report';
import SettingsIcon from '@material-ui/icons/Settings';

import { RouteItem } from './Base';

import Accessibility from '../views/Accessibility/Accessibility';
import ViewOne from '../views/ViewOne/ViewOne';
import ViewTwo from "../views/ViewTwo/ViewTwo";
import ReportView from "../views/ReportView/ReportView";
import ConfigView from '../views/ConfigView/ConfigView';

const dashboardRoutes:RouteItem[]=[
    {
        path:'/dashboard/accessibility',
        sidebarName:"accessibility.sidebar",
        navbarName:"accessibility.navbar",
        icon: AccessibilityIcon,
        component: Accessibility,
    },
    {
        path:'/dashboard/view-one',
        sidebarName:"viewOne.sidebar",
        navbarName:"viewOne.navbar",
        icon: BusinessIcon,
        component: ViewOne,
    },
    {
        path:'/dashboard/view-two',
        sidebarName:"viewTwo.sidebar",
        navbarName:"viewTwo.navbar",
        icon: ChatIcon,
        component: ViewTwo,
    },
    {
        path:'/dashboard/config',
        sidebarName:"configuration.sidebar",
        navbarName:"configuration.navbar",
        icon: SettingsIcon,
        component: ConfigView,
    },
    {
        path:'/dashboard/report',
        sidebarName:"report.sidebar",
        navbarName:"report.navbar",
        icon: ReportIcon,
        component: ReportView,
    },
];

export default dashboardRoutes;