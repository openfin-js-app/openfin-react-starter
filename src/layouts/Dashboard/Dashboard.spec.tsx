import * as React from 'react';
import { Provider } from 'react-redux';
import { createShallow, createMount } from '@material-ui/core/test-utils';
import configurestore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';

import Snackbar from '@material-ui/core/Snackbar';

import Dashboard from './Dashboard';
import GlobalContext from '../../GlobalContext';
import ViewOne from '../../views/ViewOne/ViewOne';
import {Sidebar, Header, SnackbarContent } from '../../components';
import {MuiTheme} from "../../reduxs";

const mockStore = configurestore();
const initialState = {
    application:{
        username:'',
        computerName:'',
        machineId:null,
        deviceUserId:null,
        loading:true,
        drawerOpen:true,
        launchBarCollapse:false,
        snackBarOpen:false,
        snackBarMsgInfo:{},
        snackBarMsgQueue:[],
        openfinVersion:'n/a',
        openfinHostSpec:{},
        windowsState:'normal',
    },
    client:{
        count:0,
    },
    config:{
        application:{
            theme:MuiTheme.LIGHT,
        }
    }
};

declare const window:any;

describe('Dashboard layout',()=>{

    let shallow;
    let mount;

    beforeEach(() => {
        mount = createMount();
        shallow = createShallow();
    });

    afterEach(()=>{
        mount.cleanUp();
    });

    it('render in normal state correctly by default',()=>{
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard/view-one']}>
                    <GlobalContext
                        config={initialState.config}
                    >
                        <Dashboard location={{pathname:'/dashboard/view-one'}}/>
                    </GlobalContext>
                </MemoryRouter>
            </Provider>

        );
        expect(wrapper.find(ViewOne)).toHaveLength(1);
        expect(wrapper.find(Dashboard)).toHaveLength(1);
        wrapper.find(Header).props().handleDrawerToggle();
        wrapper.find(Header).props().onSwitchToLaunchBar();
        wrapper.find(Header).props().onMinimize();
        wrapper.find(Header).props().onMaximize();
        wrapper.find(Header).props().onClose();
        // snackbar events
        expect(wrapper.find(Snackbar)).toHaveLength(1);
        wrapper.find(Snackbar).props().onClose();
        wrapper.find(Snackbar).props().onExited();

        expect(store.getActions()).toMatchSnapshot();
    })

    it('render SnackbarContent and could be closed correctly',()=>{
        const initialState = {
            application:{
                username:'',
                computerName:'',
                machineId:null,
                deviceUserId:null,
                loading:true,
                drawerOpen:true,
                launchBarCollapse:false,
                snackBarOpen:true,
                snackBarMsgInfo:{
                    variant:'primary',
                    message:'message',
                },
                snackBarMsgQueue:[],
                openfinVersion:'n/a',
                openfinHostSpec:{},
                windowsState:'normal',
            },
            client:{
                count:0,
            },
            config:{
                application:{
                    theme:MuiTheme.DARK,
                }
            }
        };

        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter initialEntries={['/dashboard/view-one']}>
                    <GlobalContext
                        config={initialState.config}
                    >
                        <Dashboard location={{pathname:'/dashboard/view-one'}}/>
                    </GlobalContext>
                </MemoryRouter>
            </Provider>
        );
        expect(wrapper.find(ViewOne)).toHaveLength(1);
        expect(wrapper.find(Dashboard)).toHaveLength(1);
        wrapper.find(Header).props().handleDrawerToggle();
        wrapper.find(Header).props().onSwitchToLaunchBar();
        wrapper.find(Header).props().onMinimize();
        wrapper.find(Header).props().onMaximize();
        wrapper.find(Header).props().onClose();
        // SnackbarContent events
        expect(wrapper.find(SnackbarContent)).toHaveLength(1);
        wrapper.find(SnackbarContent).props().onClose();

        expect(store.getActions()).toMatchSnapshot();
    })

});

