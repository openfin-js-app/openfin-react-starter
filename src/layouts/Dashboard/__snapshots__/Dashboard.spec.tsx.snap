// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard render SnackbarContent and could be closed correctly 1`] = `
Array [
  Object {
    "type": "APPLICATION_DRAWER_TOGGLE",
  },
  Object {
    "type": "APPLICATION_LAUNCH_BAR_TOGGLE",
  },
  Object {
    "error": false,
    "payload": Object {
      "callback": [Function],
      "errorCallback": [Function],
    },
    "type": "REDUX_OPENFIN::MINIMIZE",
  },
  Object {
    "type": "APPLICATION_TOGGLE_WINDOW_STATE",
  },
  Object {
    "error": false,
    "payload": Object {
      "callback": [Function],
      "errorCallback": [Function],
      "force": false,
    },
    "type": "REDUX_OPENFIN::CLOSE",
  },
  Object {
    "payload": Object {
      "open": false,
    },
    "type": "APPLICATION_SET_SNACKBAR_STATUS",
  },
]
`;

exports[`Dashboard render in normal state correctly by default 1`] = `
Array [
  Object {
    "type": "APPLICATION_DRAWER_TOGGLE",
  },
  Object {
    "type": "APPLICATION_LAUNCH_BAR_TOGGLE",
  },
  Object {
    "error": false,
    "payload": Object {
      "callback": [Function],
      "errorCallback": [Function],
    },
    "type": "REDUX_OPENFIN::MINIMIZE",
  },
  Object {
    "type": "APPLICATION_TOGGLE_WINDOW_STATE",
  },
  Object {
    "error": false,
    "payload": Object {
      "callback": [Function],
      "errorCallback": [Function],
      "force": false,
    },
    "type": "REDUX_OPENFIN::CLOSE",
  },
  Object {
    "payload": Object {
      "event": undefined,
      "reason": undefined,
    },
    "type": "APPLICATION_CLOSE_SNACKBAR",
  },
  Object {
    "type": "APPLICATION_PROCESS_SNACKBAR_QUEUE",
  },
]
`;
